File: assoc_arrays.php
<?php
    // Initialize
    $translation = array("hello"=>"merhaba",     "goodbye"=>"hoscakal");
    print("-Array after initialization: \n");
    print_r($translation);
    
    // Get the value for a given key
    print("-The value of key 'hello': ".$translation["hello"]);
    
    // Add a new element
    $translation["one"] = 1;
    print("\n-Array after 'one' is added:\n");
    print_r($translation);
    
    // Remove an element
    unset($translation["one"]);
    print("-Array after deletion of 'one':\n");
    print_r($translation);
    
    // Modify the value of an existing element
    $translation["hello"] = "selam";
    print("-Array after 'hello' => 'selam' instead of 'hello':\n");
    print_r($translation);
    
    // Search for the existence of a key
    print("\n-For the below, blank if false, 1 if true\n");
    print("The key 'hello' exists: ".array_key_exists("hello", $translation));
    
    // Search for the existence of a value
    print("\nThe value 'selam' exists: ".in_array("selam", $translation));
    
    // Loop through the array
    print("\n\n-The array printed by iteration:\n");
    foreach($translation as $key => $value) {
        foo($key, $value);
    }
    
    function foo($a, $b) {
        print("\n".$a.": ".$b);
    }
    
?>

File: assoc_arrays.dart
void main() { 
  // Initialization
  var translation = {
    'hello': 'merhaba',
    'goodbye': 'hoscakal'
  };
  
  // Get the value for a given key
  print("\nValue of key 'goodbye':");
  print(translation['goodbye']);
  
  // Add a new element
  translation['one'] = 'bir';
  print("\nThe hash after 'one' added:");
  print(translation);
  
  // Remove an element
  translation.remove('one');
  print("\nThe hash after 'one' removed:");
  print(translation);
  
  // Modify the value of an existing element
  translation['hello'] = 'selam';
  print("\nThe hash after value of 'hello' changed to 'selam':");
  print(translation);
  
  // Search for the existence of a key
  print("\nThe hash contains key 'hello':");
  print(translation.containsKey('hello'));
  print("\nThe hash contains key 'one':");
  print(translation.containsKey('one'));
  
  // Search for the existence of a value
  print("\nThe hash contains value 'selam':");
  print(translation.containsValue('selam'));
  
  print("\nThe hash contains value 'hayir':");
  print(translation.containsValue('hayir'));
  
  // Iterate through the associative array,
  // applying the function foo
  print("\nThe hash printed using iteration:");
  translation.forEach((k,v) => foo(k, v));
}

void foo(var a, var b)
{
  print(a + ": " + b);
}
